### PR TITLE
Added switch -P|--port to make the port to connect to configureable.

### PR DESCRIPTION
**Pull request by ccztux**

### CODE REVIEW

**Comment by rhagman** [comment]
Nice, I like it!
But it seems like you forgot to add a test for this new feature. : )

**Comment by ccztux** [comment]
Can you give me a hint, where i can ad such a test?

**Comment by ccztux** [comment]
BTW: The bugfix: 8e8a783357f3b6cb99b6cea2c107ee9fe01b86b1 is really important. In case that there is an issue with the session the check exits: with CRITICAL - The session is not authenticated.

**Comment by rhagman** [comment]
I think a good way of testing the added port switch would be to run the plugin with the parameters that you would like to have and adding the `--generate_test` at the end. This generates a file that you can put into `t/series` which will be run when running `make test`.

I'm not a testing expert so I dug around the better part of yesterday to see if I could find a nice way of unit testing this. I was thinking about a nice way to start to unit test part of the code and found that the best way could be to use modulinos (http://perltricks.com/article/107/2014/8/7/Rescue-legacy-code-with-modulinos/).

Thanks for the heads up about the bugfix.

File: check_vmware_api.pl [hunk 1]
```diff
@@ -377,6 +377,13 @@ sub main {
 		. '   Virtual machine name.',
 		required => 0,
 	);
+	
+	$np->add_arg(
+		spec => 'port|P=i',
+		help => "-P, --port=<port>\n"
+		. '   Port to connect to.',
+		required => 0,
```
Comment (rhagman) [inline check_vmware_api.pl]: Here you can add the default value to the argument with "default => 443,".

File: check_vmware_api.pl [hunk 2]
```diff
@@ -377,6 +377,14 @@ sub main {
 		. '   Virtual machine name.',
 		required => 0,
 	);
+	
+	$np->add_arg(
+		spec => 'port|P=i',
+		help => "-P, --port=<port>\n"
+		. "   Port to connect to.\n"
+		. '   By default, port 443 will be used.',
+		required => 0,
```
Comment (rhagman) [inline check_vmware_api.pl]: Here you can add the default value with "default => 443," then you do not need to test for it on line 551.

**Comment by ccztux** [comment]
Thank you for this advice.

**Comment by jacobbaungard** [comment]
This plugin is "deprecated" from our side, and so most likely won't see any updates. We are therefore closing all open PRs. Sorry about that.

The following project, which was forked from this one, might be of interest to you: https://github.com/BaldMansMojo/check_vmware_esx/

**Pull request closed without merging**